src/clib/pio_rearrange.c

```diff
@@ -332,9 +338,16 @@ int create_mpi_datatypes(MPI_Datatype mpitype, int msgcnt,
         if (mcount[i] > 0)
         {
             int len = mcount[i] / blocksize;
-            int displace[len];
+            int *displace = NULL;
```

Comment:
I'm not sure this change makes any sense.  First len can never be < 1, second suppose that len were == 0 you would have a bug in the following code where displace is used and freed.